croud/session.py

```diff
@@ -65,22 +65,21 @@ def __init__(
     ) -> JsonDict:
         url = self.url + endpoint
         resp = await self.client.post(
-            url, json={"query": query, "variables": variables}
+            url, json={"query": query, "variables": variables}, allow_redirects=False
         )
-        if resp.status == 200:
-            try:
-                result = await resp.json()
-                return result
-            except ContentTypeError:
-                print_error(
-                    "Unauthorized. Use `croud login` to login to CrateDB Cloud."
-                )
-                await self.client.close()
-                exit(1)
-        else:
-            raise Exception(
-                f"Query failed to run by returning code of {resp.status}. {query}"
-            )
+
+        if resp.status == 302:  # login redirect
+            print_error("Unauthorized. Use `croud login` to login to CrateDB Cloud.")
+            await self.client.close()
+            exit(1)
+
+        if resp.status != 200:
+            print_info(f"Query failed to run by returning code of {resp.status}.")
```

Comment:
I am not 100% sure whether we should print info or error here.
What's your opinion @alijared ?

Reply:
I'd go with "error". A user expected this command to work. It didn't. That's an error from the perspective of a user.

Also, the query didn't fail because of the status code, but the status code is an error code because the query failed to run:
```suggestion
            print_error(f"Query failed to run and returned an error code {resp.status}.")
```

I think it would then make sense to look if `resp.json()["errors"]` exists and may format them "nicely".

Reply:
This happens already when the query (response or error) is printed: https://github.com/crate/croud/blob/master/croud/gql.py#L75